src/main/java/org/zalando/nakadi/controller/SubscriptionStreamController.java

```diff
@@ -135,6 +143,13 @@ public StreamingResponseBody streamEvents(
             SubscriptionStreamer streamer = null;
             final SubscriptionOutputImpl output = new SubscriptionOutputImpl(response, outputStream);
             try {
+                if  (subscriptionDbRepository.getSubscription(subscriptionId).getEventTypes().stream()
+                        .map(etName -> floodService.isConsumptionBlocked(etName)).findFirst().orElse(false)) {
+                    response.setStatus(HttpStatus.TOO_MANY_REQUESTS.value());
```

Comment:
Right now, check for blocking implemented in the fail-open manner. From my point of view, it is important not to stop serving clients are not under block and if the problem with so called rate limiter other clients should not be affected by that instead we have to fix the issues faster.